src/components/cardProject.astro

```diff
@@ -0,0 +1,52 @@
+---
+interface Props {
+  img: {
+    classProp: string;
+    src: string;
+    alt: string;
+  };
+  title: string;
+  description: string;
+  skills: Array<string>;
+}
+
+const {
+  img: { classProp, src, alt },
+  title,
+  description,
+  skills,
+} = Astro.props;
+---
```

Comment:
between `---` you can write js/ts which will be used below in the template (check astro docs)